### ISSUE TITLE
suggested alternative challenge to section 7

### ISSUE THREAD

**Issue by jpiaskowski** [swcarpentry/r-novice-gapminder#461]
I'm not terribly fond of the challenges in section 7 because that particular challenges can be completed much more efficiently by dplyr functions or aggregate(). This can result in some users viewing for loops and other controls features as not useful or finding earlier parts of the workshop a waste of time. Here is one suggested change that incorporates principles taught in the workshop:

Determine for each country/year combination in Asia the change in life expectancy for the continent of Asia when that country/year combination is omitted.

Solution:

`
Asia <- gapminder[gapminder$continent == "Asia",]
mean_lifeExp <- list[[]]

for(i in 1:nrow(Asia) {
  mean_lifeExp[[i]] <- mean(Asia[-i, "lifeExp"], na.rm = T)
}
`

This can be expanded by asking that control features to provide an NA for the mean when the ith data point is missing indicating no change in the mean due to omitting the ith data point. Or having users plot the results by the index. Or output the year, country and mean as one data frame. Or pre-allocate the list to the length of object 'Asia'.

**Comment by jcoliver** [comment]
Thanks for these suggestions, @jpiaskowski . This is related to a larger issue involving tidyverse packages (see #365 and #442 ). I think it's fair to say that much of what we introduce in the first few episodes can be done more efficiently by other packages, especially when dealing with `if` conditionals and `for` loops. Are there specific challenges that you think need to be revised to better reinforce the material of episode 7?

**Comment by jpiaskowski** [comment]
I addressed this partially in issue [#460](https://github.com/swcarpentry/r-novice-gapminder/issues/460) . Another exercise could be to create an `if` command for installing and loading packages on systems when it is not known if the package is installed or not. This is a common usage of `if()` not supplanted by dplyr. The commands `while`, `repeat`, etc are trickier for the new student. I mocked up a code for the iterative process median polish to illustrate how `while` commands can be useful for iteration when a value from a previous iteration needs to be incorporated. However, I suspect a challenge like that can only be accomplished by a more advanced student within the time provided. Let me know if you think any of this is useful enough to build into a new pull request.
